addons/hr_expense/tests/test_expenses.py

```diff
@@ -926,14 +926,14 @@ def test_attachments_in_move_from_company_expense(self):
 
         self.assertRecordValues(expense_sheet.account_move_ids[0].attachment_ids, [{
             'raw': b"R0lGODdhAQABAIAAAP///////ywAAAAAAQABAAACAkQBADs=",
-            'name': 'file1.png',
+            'name': 'file2.png',
```

Comment:
The reason for this is because `_order = "date desc, id desc"` for expenses.
Se when we create the account moves iterating on `expense_line_ids`
```python
        payments = self.env['account.payment'].with_context(**skip_context).create([
            expense._prepare_payments_vals() for expense in company_account_sheets.expense_line_ids
        ])
        moves |= payments.move_id
        moves.action_post()
```

We actually get them in reverse order.
Then when we read `expense_sheet.account_move_ids`, we keep that order because the `name` was computed by keeping that order in `action_post`.

It might have worked before only because there is now an invalidation that didn't happen before and we kept the order of `expense_line_ids` as set in the `create`

Reply:
Using indexes like this **and syncing attachments manually between expenses, sheets AND moves** :exploding_head: is really not a good idea.

Reply:
might be nice to assert which move is from which expense to make it obvious that everything is working correctly

Reply:
Might be even better to not do things that lead to the need of writing a test like this when it is perfectly avoidable...

Reply:
Agree but we have to deal with what's there until we clean it